src/module_3/module_3_milestone2.py

```diff
@@ -0,0 +1,104 @@
+import pandas as pd
+import numpy as np
+from sklearn import linear_model
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScaler
+from sklearn import metrics
+import joblib
+
+
+DATA_PATH = "/home/unai/datasets/feature_frame.csv"
+training_columns = [
+    "abandoned_before",
+    "ordered_before",
+    "global_popularity",
+    "user_order_seq",
+    "normalised_price",
+]
+label_col = "outcome"
+model_filename = "groceries_model.pkl"
+
+
+def main():
+    data = load_data(DATA_PATH)
+    # processed_data = preprocessing(data)
+    # model = training_model(processed_data)
+    # save_model(model, model_filename)
+    model = load_model(model_filename)
+    threshold = threshold_calculation(data, model)
+    # with these 3 variables we can predict new values with the model_prediction function
+
+
+def load_data(data_path: str) -> pd.DataFrame:
+    return pd.read_csv(data_path)
+
+
+def preprocessing(data: pd.DataFrame) -> pd.DataFrame:
+    # We only keep orders with at least 5 items
+    orders = data["order_id"].unique()
+    orders_more5 = orders[data.groupby("order_id")["outcome"].sum() >= 5]
+    return data.loc[data["order_id"].isin(orders_more5)]
+
+
+def training_model(data: pd.DataFrame) -> make_pipeline:
```

Comment:
[make_pipeline](https://scikit-learn.org/stable/modules/generated/sklearn.pipeline.make_pipeline.html) returns a Pipeline not a make_pipeline